src/main.ts

```diff
@@ -1,13 +1,42 @@
 import citiesJson from './json_data/citiesJson.json';
-import { initializeMap, addPinsToMap, addCustomPindropFunctionality } from './map.ts';
-import { addListButtonClickListener, addCloseButtonClickListener } from './domManipulation.ts';
+import { initializeMap, addPinsToMap, dropPinClickCallback, mapSetViewCallback } from './map.ts';
+import { addListButtonClickListener, addCloseButtonClickListener, addDropPinClickListener, populateCurrentWeather, populateHourlyWeather, toggleWeatherContainer, cityFunctionality } from './domManipulation.ts';
+import getWeatherData from './apiHelper.ts';
 import { CityInterface } from './interfaces';
 import './style.css';
 
 const map = initializeMap();
 const cities: CityInterface[] = citiesJson;
-addPinsToMap(map, cities);
+let customButtonOn = false; 
+
+addPinsToMap(map, cities, setWeather);
+addPinsFunctionality(map);
 addCustomPindropFunctionality(map);
 
 addListButtonClickListener();
-addCloseButtonClickListener();
\ No newline at end of file
+addCloseButtonClickListener();
+
+function addCustomPindropFunctionality(map: L.Map): void {
+  addDropPinClickListener(() => {
+    customButtonOn = !customButtonOn;
+    dropPinClickCallback(map, customButtonOn, setWeather);
+  });
+} 
+
+async function setWeather(city: CityInterface): Promise<void> {
+  toggleWeatherContainer();
+  try {
+      const [currentWeatherData, hourlyWeatherData] = await getWeatherData(city);
+      populateCurrentWeather(currentWeatherData, city);
+      populateHourlyWeather(hourlyWeatherData, currentWeatherData);
+      toggleWeatherContainer();
+  } catch (error) {
+      throw error;
```

Comment:
At this point, nothing but the browser is catching this.
Handle it yourself with a `console.error` or some popup to the user so they know something bad happened